### PR TITLE
implementing numpy frontend extract function

### PR DESCRIPTION
**Pull request by Nuno-Ferreira**
solves issue #10661

### CODE REVIEW

File: ivy/functional/frontends/numpy/sorting_searching_counting/searching.py [hunk 1]
```diff
@@ -92,3 +92,8 @@ def nanargmax(a, /, *, axis=None, out=None, keepdims=False):
 def nanargmin(a, /, *, axis=None, out=None, keepdims=False):
     a = _nanargminmax(a, axis=axis)
     return ivy.argmin(a, axis=axis, keepdims=keepdims, out=out)
+
+
+@to_ivy_arrays_and_back
+def extract(condition, arr):
+    return ivy.extract(condition, arr)
```
Comment (jkeane508) [inline ivy/functional/frontends/numpy/sorting_searching_counting/searching.py:99]: Ivy doesn't have an extract function in its library. This function would have to be implemented as a combination of other Ivy functions

File: ivy_tests/test_ivy/test_frontends/test_numpy/test_sorting_searching_counting/test_searching.py [hunk 2]
```diff
@@ -343,3 +343,31 @@ def test_numpy_nanargmin(
         axis=axis,
         keepdims=keep_dims,
     )
+
+# extract
+@handle_frontend_test(
+    fn_tree="numpy.extract",
+    dtype_x_condition=helpers.dtype_values_condition(
```
Comment (jkeane508) [inline ivy_tests/test_ivy/test_frontends/test_numpy/test_sorting_searching_counting/test_searching.py:350]: Test logs are saying that `dtype_values_condition` doesn't exist

**Review by jkeane508 — CHANGES_REQUESTED** [review]
Hey,

Thanks for submitting this PR

Seems to be lint errors in the files changed in this PR, you can install a pre-commit hook to prevent these lint errors in advance (see the setting up section of the contributor docs)

There also seem to be some errors with your implementation - see below

Happy Coding! :)

**Comment by Nuno-Ferreira** [comment]
Hi James,

Thank you for reviewing it so quickly!

I will work on it as soon as I can and make the necessary changes.

**Comment by Nuno-Ferreira** [comment]
Hey @jkeane508, 

I thought of implementing the`.where` function to use but I was wondering if there would be a problem because the `.where` function has more arguments, what is your opinion?

I tried to run pytest in my container but wasn't able to since it just throws an error:
root@3396fdf3143a:/workspaces/ivy# pytest /home/nuno/Documents/Ivy/ivy_tests/test_ivy/test_frontends/test_numpy/test_sorting_searching_counting::test_searching
================================================================ test session starts =================================================================
platform linux -- Python 3.8.10, pytest-7.1.2, pluggy-1.0.0
rootdir: /workspaces/ivy
plugins: metadata-2.0.4, typeguard-2.13.3, hypothesis-6.55.0, json-report-1.5.0
collected 0 items                                                                                                                                    

=============================================================== no tests ran in 0.00s ================================================================
ERROR: file or directory not found: /home/nuno/Documents/Ivy/ivy_tests/test_ivy/test_frontends/test_numpy/test_sorting_searching_counting::test_searching

Hope this is better! Let me know if everything is ok!

Happy coding! :)

**Comment by Nuno-Ferreira** [comment]
Hey James,

Thank you very much for your advice! I got it to work!

Just to clarify, when you say implement `extract()` do you mean implement it as a an Ivy function? Like `ivy.where()`?

**Comment by jkeane508** [comment]
> 

No, not exactly, the task is to implement `numpy.extract()` in our Ivy NumPy frontend so this means that it wouldn't exist in the ivy namespace, `ivy.x()`. Our Ivy frontends are used to take in functions from external frameworks and "wrap" them using ivy functions. We currently do not have an implementation of `ivy.extract()` and we aren't intending on adding any more functions to the Ivy functional API right now. Protocol for when a frontend function doesn't directly exist in Ivy is to implement it using a combination of already existing Ivy functions. You can read more about it [here](https://lets-unify.ai/ivy/deep_dive/ivy_frontends.html)

Good Luck!

**Pull request closed without merging**